### PR TITLE
Route matching in short trips with small movements returns valid route

### LINKED ISSUE #66: Route matching in short trips with small movements
**Problem**

In some short trips, in which the car makes small movements, and therefore the coordinates vary a little, no valid match is obtained from the route match.

**Expected behaviour**

A valid match is obtained from  short trips with small movements.

### PR DESCRIPTION
**Pull request by pierDipi**
**Improvements**

In some short trips, in which the car makes small movements, and
therefore the coordinates vary a little, a valid match is obtained
from the route match.

**Related issue**

Closes #66

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -20,6 +20,7 @@ buildNumber.properties
 .mvn/wrapper/maven-wrapper.jar
 
 src/test/resources/graphHopperSource/*
-src/test/resources/graphHopper/*
+src/test/resources/graphhopper/*
+src/test/resources/osm/maps/germany
```
Comment (andr3a87) [inline .gitignore:24]: it is necessary?

File: src/test/resources/simplelogger.properties [hunk 2]
```diff
@@ -1,4 +1,4 @@
-org.slf4j.simpleLogger.defaultLogLevel=info
+org.slf4j.simpleLogger.defaultLogLevel=debug
```
Comment (andr3a87) [inline src/test/resources/simplelogger.properties:1]: revert to info

File: src/main/scala/it/agilelab/bigdata/gis/domain/graphhopper/GraphHopperManager.scala [hunk 3]
```diff
@@ -149,13 +101,74 @@ case class GraphHopperManager(conf: Config) extends RouteMatcher with Logger {
...
+        })
+
+    })
+  }
+
+  private def getMappedEdges(edges: Seq[EdgeMatch]) = {
+    edges
+      .map(edge => (graphConf.encoder.getHighwayAsString(edge.getEdgeState), edge.getEdgeState.getDistance))
+      .map {
+        case (null, value) => ("unclassified", value)
```
Comment (andr3a87) [inline src/main/scala/it/agilelab/bigdata/gis/domain/graphhopper/GraphHopperManager.scala:161]: null is returned when an uncensored value is found in the encoder. Can we add a warn log?

File: src/test/resources/simplelogger.properties (hunk 2 above)
Comment (andr3a87) [inline src/test/resources/simplelogger.properties:1]: revert to info

File: src/main/scala/it/agilelab/bigdata/gis/domain/graphhopper/GraphHopperManager.scala [hunk 4]
```diff
@@ -149,13 +101,74 @@ case class GraphHopperManager(conf: Config) extends RouteMatcher with Logger {
...
+        case x: (String, Double) => x
+      }
+  }
+}
+
+object GraphHopperManager {
+
+  private def getRouteTypesKm(mappedEdges: Seq[(String, Double)]) = {
+    mappedEdges
+      .groupBy(_._1)
```
Comment (andr3a87) [inline src/main/scala/it/agilelab/bigdata/gis/domain/graphhopper/GraphHopperManager.scala:171]: can we use a name of variable to improve the readability?
```
.groupby{case (name1,name2) => ... }
.map {case (name1,.....) => ... }
````
maybe we can use also type alias

File: src/main/scala/it/agilelab/bigdata/gis/domain/graphhopper/GraphHopperManager.scala [hunk 5]
```diff
@@ -149,13 +101,74 @@ case class GraphHopperManager(conf: Config) extends RouteMatcher with Logger {
...
+      .map {
+        case (null, value) => ("unclassified", value)
+        case x: (String, Double) => x
+      }
+  }
+}
+
+object GraphHopperManager {
+
+  private def getRouteTypesKm(mappedEdges: Seq[(String, Double)]) = {
```
Comment (andr3a87) [inline src/main/scala/it/agilelab/bigdata/gis/domain/graphhopper/GraphHopperManager.scala:169]: add scaladoc

File: src/main/scala/it/agilelab/bigdata/gis/domain/graphhopper/GraphHopperManager.scala (hunk 3 above)
Comment (pierDipi) [reply src/main/scala/it/agilelab/bigdata/gis/domain/graphhopper/GraphHopperManager.scala:161]: Added in `graphConf.encoder.getHighwayAsString` to provide the necessary context to the warning.

**Review by andr3a87 — APPROVED** [review]

**Pull request merged**